src/metabase/api/common.clj

```diff
@@ -255,6 +256,51 @@
                                   (ns-name *ns*) fn-name)))
       (assoc parsed :fn-name fn-name, :route route, :docstr docstr))))
 
+(def ^:private content-type->regex
+  "A map of content type to a regex that should match that content type."
+  {:content/json #"^application/(.+\+)?json"
+   :content/form #"(^application/x-www-form-urlencoded)|(multipart/form-data)|(application/form)"
+   :content/*    (constantly true)})
```

Comment:
```suggestion
   :content/*    #".*"})
```

nit: we can do this so the `content-type-matches` don't have to deal with the fact that `value` of `content-type->regex` could be a func or a regex?

Reply:
I was originally going to do this. The problem is that often requests will not have a content type so it would be a regex against a nil string. Changing to a function that just constantly returns true is exactly what we want, regardless of the content-type or lack of it.

Reply:
if we really want uniformity, perhaps the regexes could just be partial'd so that all of the matchers are just functions.